src/RequestHandle.cpp

```diff
@@ -480,4 +553,64 @@ auto curl_write_data(
     return data_length;
 }
 
+auto curl_xfer_info(
+    void *p
+) -> int
+{
+    auto* raw_request_ptr = static_cast<RequestHandle*>(p);
+
+    /**
+     * If WroteMaxBytes is true, we can terminate this request.
+     * Returning 1 will cause the cURL request to be aborted.
+     * (Note: This will result in the CURLcode returning as an error.
+     *  This is handled in setCompletionStatus.)
+     **/
+    if (raw_request_ptr->GetCompletedWritingMaxBytes())
```

Comment:
I would suggest since this is a friend function calling into a private function of the RequestHandle so it can manipulate the max download bytes fields -- they are friends intentionally to keep the API very simple for the end user and by making these public you are leaking internal RequestHandle details through the public API

Reply:
Makes sense! Changed to private functions.